error: attempt to use a poisoned identifier

*   Version of collectd: 5.8.0
*   Operating system / distribution:
FreeBSD 11.1 amd64
MariaDB 10.2.14
FreeBSD clang version 4.0.0 (tags/RELEASE_400/final 297347) (based on LLVM 4.0.0)

## Expected behavior

Build successful

## Actual behavior

Build fails with error
```
libtool: link: ( cd ".libs" && rm -f "memory.la" && ln -s "../memory.la" "memory.la" )
/bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I./src  -I./src -I./src/daemon -DPREFIX='"/usr/local"' -DCONFIGFILE='"/usr/local/etc/collectd.conf"' -DLOCALSTATEDIR='"/var"' -DPKGLOCALSTATEDIR='"/var/db/collectd"' -DPLUGINDIR='"/usr/local/lib/collectd"' -DPKGDATADIR='"/usr/local/share/collectd"' -I/usr/local/include -Wall  -I/usr/local/include/mysql -I/usr/local/include/mysql/mysql -I/usr/local/include -O2 -fno-strict-aliasing -pipe -march=native  -fstack-protector -MT src/mysql_la-mysql.lo -MD -MP -MF src/.deps/mysql_la-mysql.Tpo -c -o src/mysql_la-mysql.lo `test -f 'src/mysql.c' || echo './'`src/mysql.c
libtool: compile:  cc -DHAVE_CONFIG_H -I. -I./src -I./src -I./src/daemon -DPREFIX=\"/usr/local\" -DCONFIGFILE=\"/usr/local/etc/collectd.conf\" -DLOCALSTATEDIR=\"/var\" -DPKGLOCALSTATEDIR=\"/var/db/collectd\" -DPLUGINDIR=\"/usr/local/lib/collectd\" -DPKGDATADIR=\"/usr/local/share/collectd\" -I/usr/local/include -Wall -I/usr/local/include/mysql -I/usr/local/include/mysql/mysql -I/usr/local/include -O2 -fno-strict-aliasing -pipe -march=native -fstack-protector -MT src/mysql_la-mysql.lo -MD -MP -MF src/.deps/mysql_la-mysql.Tpo -c src/mysql.c  -fPIC -DPIC -o src/.libs/mysql_la-mysql.o
In file included from src/mysql.c:36:
/usr/local/include/mysql/mysql.h:139:5: error: attempt to use a poisoned identifier
    strcpy((a)->net.sqlstate, "00000");\
    ^
In file included from src/mysql.c:36:
In file included from /usr/local/include/mysql/mysql.h:423:
/usr/local/include/mysql/mariadb_stmt.h:47:3: error: attempt to use a poisoned identifier
  strcpy((a)->sqlstate, "00000");\
  ^
2 errors generated.
gmake[3]: *** [Makefile:6107: src/mysql_la-mysql.lo] Error 1
gmake[3]: Leaving directory '/usr/ports/net-mgmt/collectd5/work/collectd-5.8.0'
gmake[2]: *** [Makefile:4169: all] Error 2
gmake[2]: Leaving directory '/usr/ports/net-mgmt/collectd5/work/collectd-5.8.0'
*** Error code 1

Stop.
make[1]: stopped in /usr/ports/net-mgmt/collectd5
*** Error code 1
```

## Steps to reproduce

*   Build/install MariaDB 10.2 client
*   configure `--with-libmysql=/usr/local`
*   make

--- a/src/daemon/collectd.h
+++ b/src/daemon/collectd.h
@@ -240,26 +240,6 @@
 #define __attribute__(x) /**/
 #endif
 
-#if defined(COLLECT_DEBUG) && COLLECT_DEBUG && defined(__GNUC__) && __GNUC__
-#undef strcpy
-#undef strcat
-#undef strtok
-#pragma GCC poison strcpy strcat strtok
-#endif
-
-/*
- * Special hack for the perl plugin: Because the later included perl.h defines
- * a macro which is never used, but contains `sprintf', we cannot poison that
- * identifies just yet. The parl plugin will do that itself once perl.h is
- * included.
- */
-#ifndef DONT_POISON_SPRINTF_YET
-#if defined(COLLECT_DEBUG) && COLLECT_DEBUG && defined(__GNUC__) && __GNUC__
-#undef sprintf
-#pragma GCC poison sprintf
-#endif
-#endif
-
 #ifndef GAUGE_FORMAT
 #define GAUGE_FORMAT "%.15g"
 #endif

--- a/src/lua.c
+++ b/src/lua.c
@@ -28,26 +28,17 @@
  *   Ruben Kerkhof <ruben at rubenkerkhof.com>
  **/
 
-/* <lua5.1/luaconf.h> defines a macro using "sprintf". Although not used here,
- * GCC will complain about the macro definition. */
-#define DONT_POISON_SPRINTF_YET
-
+#include "collectd.h"
 #include "common.h"
 #include "plugin.h"
-#include "collectd.h"
+#include "utils_lua.h"
 
 /* Include the Lua API header files. */
 #include <lauxlib.h>
 #include <lua.h>
 #include <lualib.h>
-#include "utils_lua.h"
 
 #include <pthread.h>
-
-#if COLLECT_DEBUG && __GNUC__
-#undef sprintf
-#pragma GCC poison sprintf
-#endif
 
 typedef struct lua_script_s {
   char *script_path;

--- a/src/perl.c
+++ b/src/perl.c
@@ -33,20 +33,10 @@
 /* do not automatically get the thread specific Perl interpreter */
 #define PERL_NO_GET_CONTEXT
 
-#define DONT_POISON_SPRINTF_YET 1
-#include "collectd.h"
-
-#undef DONT_POISON_SPRINTF_YET
-
 #include <stdbool.h>
 
 #include <EXTERN.h>
 #include <perl.h>
-
-#if defined(COLLECT_DEBUG) && COLLECT_DEBUG && defined(__GNUC__) && __GNUC__
-#undef sprintf
-#pragma GCC poison sprintf
-#endif
 
 #include <XSUB.h>
 

--- a/src/utils_lua.c
+++ b/src/utils_lua.c
@@ -24,10 +24,6 @@
  *   Florian Forster <octo at collectd.org>
  **/
 
-/* <lua5.1/luaconf.h> defines a macro using "sprintf". Although not used here,
- * GCC will complain about the macro definition. */
-#define DONT_POISON_SPRINTF_YET
-
 #include "common.h"
 #include "utils_lua.h"
 

--- a/src/utils_lua.h
+++ b/src/utils_lua.h
@@ -27,12 +27,9 @@
 #ifndef UTILS_LUA_H
 #define UTILS_LUA_H 1
 
-#include "plugin.h"
 #include "collectd.h"
+#include "plugin.h"
 
-#ifndef DONT_POISON_SPRINTF_YET
-#error "Files including utils_lua.h need to define DONT_POISON_SPRINTF_YET."
-#endif
 #include <lua.h>
 
 /*

